query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala

```diff
@@ -388,3 +391,96 @@ final case class AbsentFunctionMapper(columnFilter: Seq[ColumnFilter], rangePara
 
   override def canHandleEmptySchemas: Boolean = true
 }
+
+final case class BucketValues(schema: HistogramBuckets,
+                              buckets: debox.Buffer[Array[Double]])
+
+/**
+ * Expands a Histogram RV to the equivalent Prometheus
+ * data model where we have one vector per histogram bucket.  Changes:
+ * - Each bucket converts to a vector with metric name appended with _bucket and extra le= tag
+ *
+ * If the histogram schema changes, there will be output vectors for each different bucket across all schemas.
+ * To simplify conversion, all output vectors will have the same set of timestamps.  If there is no bucket value
+ * for that timestamp (due to schema changes), then a NaN will be added.
+ *
+ * Uses up memory for each bucket of each histogram vector, one vector at a time.  Unfortunately this is memory
+ * intensive.  Due to the semantics of Observables and the lazy nature, we cannot reuse data structures as
+ * vectors may be pulled by the next reader at the same time this observable processes future vectors.
+ */
+final case class HistToPromSeriesMapper(sch: PartitionSchema) extends RangeVectorTransformer {
+  import RangeVectorTransformer._
+
+  protected[exec] def args: String = s"HistToPromSeriesMapper(options=${sch.options})"
+
+  def funcParams: Seq[FuncArgs] = Nil
+  def apply(source: Observable[RangeVector],
+            queryConfig: QueryConfig,
+            limit: Int,
+            sourceSchema: ResultSchema,
+            paramResponse: Seq[Observable[ScalarRangeVector]]): Observable[RangeVector] = {
+    // Passthrough if RV is not histogram
+    if (valueColumnType(sourceSchema) != ColumnType.HistogramColumn) return source
```

Comment:
Shouldnt we let the user know to not apply the conversion if not needed ? I would prefer to error out.

Reply:
Ok I removed it.  It's not needed anymore with refactor in http module.